### PR TITLE
Decaf

### PR DESCRIPTION
**Pull request by rodrigc**
WIP:
  1.  I merged from master
  2.  I made one fix to tests.webpack.js to only pull in spec.js files
  3.  Replace module('bbData') with angular.module('bbData')
  4.  Switched from PhantomJS to Chrome for testing

### CODE REVIEW

**Comment by rodrigc** [comment]
@uglycoyote I did `npm install buildbot-data`.
In `node_modules/buildbot-data/index.js`, I see:

```
require('./dist/buildbot-data');
module.exports = 'bbData';
```

How do I make this module visible in the `data_module` tests?

File: www/data_module/src/classes/base.service.spec.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 describe('Base class', function() {
     let $q, dataService, socketService;
-    beforeEach(module('bbData'));
+    beforeEach(angular.module('bbData'));
```
Comment (tardyp) [inline www/data_module/src/classes/base.service.spec.js]: I think module in the spec is different from angular.module:
https://docs.angularjs.org/api/ngMock/function/angular.mock.module

**Comment by uglycoyote** [comment]
@rodrigc Did you get past the ```Module 'bbData' is not available``` error?    Reading back in the comments you were referring to, it sounds like I was stuck on the same thing.

In any case, when I grabbed the latest stuff from ```tardyp:decaf``` I'm still getting that error when running ```npm test``` in the data module directory.  

I'm not clear on how to get past it, as I don't really have a great understanding of Angular's injector/module system.    

As I understand it, there needs to be an ```angular.module('bbData', ... )``` call to register ```bbData``` with the injector, prior to it being used in another module.

I notice that the file /www/base/libs/buildbot-data/dist/buildbot-data.js has a line like this:

```
  angular.module('bbData', new App());
```

However, looking at the error in Chrome (set ```singleRun:false``` in ```karma.conf.js``` to make Chrome window stay up while running ```npm test```), I can inspect the source of the webpack bundle ```tests.webpack.js```, and just searching through that file I can see that the first occurrence of ```bbData``` is the usage:

```
describe('Base class', function () {
    var $q = void 0,
        dataService = void 0,
        socketService = void 0;
    beforeEach(angular.module('bbData'));
```

so it is clear from that perspective what's causing the error.

I'm not clear on what to do to resolve the problem, but it does seem fishy that the ```data-module.js``` is declaring

```
angular.module('app', new App());
```

maybe it should be doing something more like 

```
angular.module('bbData'), new BBData());
``` 

I got the impression, looking through the sources after @tardyp  ran decaffienate, that maybe the decaffienate script likes to just call modules "App" when it's not sure what else to call them, and maybe there's a bunch of cleanup to do in fixing all of the places where it made silly mistakes like this?

Outside of the angular world, it would be completely clear to me what needs to be done -- just add an import statement.

**Comment by uglycoyote** [comment]
@rodrigc I got a few more steps along with some of these errors on decaf branch

My suggestion above of having ```data.module.js``` call ```angular.module('bbData')``` to register the class seemed to get past that error

After this I was getting

```
Uncaught Error: beforeEach expects a function argument; received [object Object]
```

which I solved by changing this line in ```base.service.spec.js```

    beforeEach(angular.module('bbData'));

to

    beforeEach(angular.mock.module('bbData'));

(this issue https://github.com/angular/angular.js/issues/7056 helped clear up my confusion about whether to use angular.module vs angular.mock.module)

after this I'm getting 

```
Uncaught Error: [$injector:unpr] Unknown provider: BaseProvider <- Base
```

Which I think is because the sources are being ordered incorrectly in the bundle, because the line with ```.factory('Base', ...)``` coming from ```base.service.js``` is appearing in the bundle but is at a greater line number than where the error is.

I know that with Typescript/Ecmascript ```import``` statements, webpack takes care of the ordering that the source files appear in the bundle, but perhaps webpack just doesn't understand the dependencies created by these angular module statements, etc, well enough to order the sources correctly?

**Comment by rodrigc** [comment]
@uglycoyote can you take any fixes that you do and file a pull request directly against the tardyp:decaf branch?

That's what I did here:

https://github.com/tardyp/buildbot/pull/31

Your ideas are good, and with 3 of us collaborating, this will help move things along.

**Comment by uglycoyote** [comment]
Sure, I'll do a pull request for what I've got soon.

I started trying to add ```require``` statements to some of the sources to get webpack to bundle them in the correct order,  though I'm not sure if that's the best approach.

**Comment by uglycoyote** [comment]
check out pull request https://github.com/tardyp/buildbot/pull/32

**Pull request merged**